core/src/main/java/io/ddf/analytics/AggregationHandler.java

```diff
@@ -146,21 +158,46 @@ public DDF agg(List<String> aggregateFunctions) throws DDFException {
     }
   }
 
-  private String convertAggregateFunctionsToSql(String sql) throws DDFException {
+  private static class AggregatedColumnExpression {
+    public final String columnName;
+    public final String columnExpression;
+
+    public AggregatedColumnExpression(String columnName, String columnExpression) {
+      this.columnName = columnName;
+      this.columnExpression = columnExpression;
+    }
+
+    public AggregatedColumnExpression(String expression) {
+      this(null, expression);
+    }
+
+    // TODO: this function need test
+    public static AggregatedColumnExpression fromString(String exp) throws DDFException {
+      if (Strings.isNullOrEmpty(exp)) {
+        throw new DDFException("Aggregation function cannot be null or empty");
+      }
 
-    if (Strings.isNullOrEmpty(sql)) return null;
+      // XXX @nhanitvn what is this regexp doing?
```

Comment:
A legacy code. I am working on expression processing for transform API too. Let's me handle this in that PR too.